### PR TITLE
Update azure-core-amqp verison and change log for Oct release.

### PR DESCRIPTION
**Pull request by YijunXieMS**

### CODE REVIEW

**Review by saragluna — COMMENTED** [review]
The servicebus sdk is now consuming an `unreleased` version of the azure-core-amqp library. And   our [java-spring](https://dev.azure.com/azure-sdk/internal/_build?definitionId=1327) nightly build pipeline keeps failing because of the version of azure-core-amqp not being bumped after the release.

File: eng/versioning/version_client.txt [hunk 1]
```diff
@@ -135,8 +135,6 @@ com.microsoft.azure:spring-cloud-starter-azure-appconfiguration-config;1.2.8-bet
 # Format;
 # unreleased_<groupId>:<artifactId>;dependency-version
 # note: The unreleased dependencies will not be manipulated with the automatic PR creation code.
-unreleased_com.azure:azure-core;1.9.0
-unreleased_com.azure:azure-core-amqp;1.6.0-beta.1
```
[the remark this answers is not in the document]
Comment (YijunXieMS) [reply eng/versioning/version_client.txt:139]: Thanks @saragluna 
I added it back and changed to 1.6.0, which we're releasing. I also updated the service dependency to 1.6.0 to pass the CI. But we're not releasing servicebus from this branch.

**Review by conniey — APPROVED** [review]

File: sdk/core/azure-core-amqp/README.md [hunk 2]
```diff
@@ -11,12 +11,12 @@ own AMQP client library that abstracts from the underlying transport library's i
 
 ### Include the package
 
-[//]: # ({x-version-update-start;com.azure:azure-core-amqp;current})
+[//]: # ({x-version-update-start;com.azure:azure-core-amqp;dependency})
```
Comment (samvaity) [inline sdk/core/azure-core-amqp/README.md:14]: @alzimmermsft Could you confirm if core packages keep this as current version?

Comment (YijunXieMS) [reply sdk/core/azure-core-amqp/README.md]: You're right. Changed back to current.

**Review by alzimmermsft — APPROVED** [review]

**Review by samvaity — APPROVED** [review]

**Review by hemanttanwar — APPROVED** [review]
LGTM

**Comment by JonathanGiles** [comment]
Just so I am clear - we do not plan to release a GA of azure-core-amqp with the BinaryData changes **removed**, correct?

**Comment by YijunXieMS** [comment]
> Just so I am clear - we do not plan to release a GA of azure-core-amqp with the BinaryData changes **removed**, correct?

@hemanttanwar You know more on this question. Could you help clarify?

**Pull request merged**